src/ImpactEvaluator.sol

```diff
@@ -94,38 +94,35 @@ contract ImpactEvaluator is AccessControl {
         uint roundIndex,
         address payable[] memory addresses,
         uint64[] memory scores,
-        string memory summaryText
+        bool moreScoresExpected
     ) public {
         require(hasRole(EVALUATE_ROLE, msg.sender), "Not an evaluator");
         require(
             addresses.length == scores.length,
             "Addresses and scores length mismatch"
         );
         require(roundIndex < currentRoundIndex(), "Round not finished");
+
         Round storage round = rounds[roundIndex];
         require(round.exists, "Round does not exist");
         require(!round.scoresSubmitted, "Scores already submitted");
-        validateScores(scores);
+        
         for (uint i = 0; i < addresses.length; i++) {
-            round.scores[addresses[i]] = scores[i];
+            round.addresses.push(addresses[i]);
+            round.scores.push(scores[i]);
         }
-        round.summaryText = summaryText;
-        round.scoresSubmitted = true;
-        reward(addresses, scores);
-    }
 
-    function validateScores(uint64[] memory scores) pure private {
-        uint64 sum = 0;
-        for (uint i = 0; i < scores.length; i++) {
-            sum += scores[i];
+        if (!moreScoresExpected) {
+            round.scoresSubmitted = true;
+            reward(round.addresses, round.scores);
         }
-        require(sum <= MAX_SCORE, "Sum of scores too big");
     }
 
     function reward(
         address payable[] memory addresses,
         uint64[] memory scores
     ) private {
+        validateScores(scores);
```

Comment:
> Validation has to be moved to `reward()` then, where it gets access to all scores

I think this is a good minimal version.

I think it would be better to run the validation on combined scores as part of ` setScores()` call too, so that we can catch the problem sooner.

In the current version:
1. First call to `setScores()` sets adds `0.5`
2. Second call to `setScores()` sets adds `0.9` - no problem noticed, although we are already over `1.0`
3. Second call to `setScores()` sets total  `0.1` and now the error is reported?

The trick is to run `validateScores` on the combination of older scores with the currently reported scores.

Thoughts?

Reply:
I don't think checking earlier is an improvement. The flow is as follows: At the end of each round, the evaluate service will calculate all scores. Afterwards, it batches the scores and then calls `setScores` multiple times. From that service's perspective, there is no measurable benefit in aborting early. Also, if you abort early, there still is no way to recover, since you already started submitting scores. I think if we have a scores overflow, we will just not pay any rewards for this round.

Reply:
I'm going to merge for now, if you disagree let's open an issue for this